edx_notifications/openedx/group_project.py

```diff
@@ -48,31 +63,25 @@ def register_notification_types(sender, **kwargs):  # pylint: disable=unused-arg
     This will be called automatically on the Notification subsystem startup (because we are
     receiving the 'perform_type_registrations' signal)
     """
-
-    register_notification_type(
-        NotificationType(
-            name=u'open-edx.xblock.group-project.file-uploaded',
-            renderer='edx_notifications.openedx.group_project.GroupProjectFileUploadedRenderer',
-        )
-    )
-
-    register_notification_type(
-        NotificationType(
-            name=u'open-edx.xblock.group-project.stage-open',
-            renderer='edx_notifications.openedx.group_project.GroupProjectStageOpenRenderer',
-        )
-    )
-
-    register_notification_type(
-        NotificationType(
-            name=u'open-edx.xblock.group-project.stage-due',
-            renderer='edx_notifications.openedx.group_project.GroupProjectStageDueRenderer',
+    mapping = {
+        NotificationMessageTypes.FILE_UPLOADED: 'GroupProjectFileUploadedRenderer',
+        NotificationMessageTypes.STAGE_OPEN: 'GroupProjectStageOpenRenderer',
+        NotificationMessageTypes.STAGE_DUE: 'GroupProjectStageDueRenderer',
+        NotificationMessageTypes.GRADES_POSTED: 'GroupProjectGradesPostedRenderer',
+    }
+
+    for message_type, renderer in mapping.iteritems():
+        register_notification_type(
+            NotificationType(
+                name=u"{prefix}.{type}".format(prefix=GROUP_PROJECT_V1_NOTIFICATION_PREFIX, type=message_type),
+                renderer="{prefix}.{renderer}".format(prefix=GROUP_PROJECT_RENDERER_PREFIX, renderer=renderer),
+            )
         )
-    )
 
-    register_notification_type(
-        NotificationType(
-            name=u'open-edx.xblock.group-project.grades-posted',
-            renderer='edx_notifications.openedx.group_project.GroupProjectGradesPostedRenderer',
+        # GP v2 can reuse GP v1 renderers
```

Comment:
@chrisndodge there are no schema changes in payloads.

The thing is, the notifications should lead to a slightly different url: namely, GPv2 uses `activate_block_id` feature to allow navigating to various types of nested blocks. This can't be resolved client-side, because `activate_block_id` is passed all the way down to the XBlock being rendered.

The reason why I had to do this is that we still need to support GP v1 with old-style links, and GP v2 with new links. This could be done via URL resolvers, but (if I do understand right) URl resolver is applied to all messages of some type. In order to apply different resolver one needs a dedicated message type. So here they are :)